### PR TITLE
Add Aoide DAC II Driver

### PR DESCRIPTION
**Pull request by howardqiao**
Add Aoide DAC II Driver based on ESS Sabre9018K2m.

http://www.aoide.cc/

### CODE REVIEW

**Comment by pelwell** [comment]
1. Overlays should be added to the Makefile in alphabetical order.
2. The overlay should be listed in the README, being careful to obey the rules (alphabetical order, same indentation, two blank lines between entries, no TABs).
3. The codec isn't a Pi-specific item and should be submitted upstream. Once it is accepted we will cherry-pick it into the current kernels.
4. scripts/checkpatch.pl finds 11 errors and 22 warnings in the commit. You will need to address these before submitting.

Otherwise it looks good.

File: arch/arm/boot/dts/overlays/README [hunk 1]
```diff
@@ -235,6 +235,15 @@ Params: 24db_digital_gain       Allow gain to be applied via the PCM512x codec
                                 that does not result in clipping/distortion!)
 
 
+Name:   Aoide DAC II
+Info:   Aoide DAC II Support
+Load:   dtoverlay=aoide-dacii,<param>=<val>
+Params: slave                   Make Aoide DAC II into slave mode, using Pi
+                                as master for bit clock and frame clock.
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: There is trailing whitespace on this line.

File: arch/arm/boot/dts/overlays/README [hunk 2]
```diff
@@ -235,6 +235,15 @@ Params: 24db_digital_gain       Allow gain to be applied via the PCM512x codec
                                 that does not result in clipping/distortion!)
 
 
+Name:   Aoide DAC II
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README]: The name should be aoide-dacii (to match the overlay), but actually both should be aoide-dacii-audio (this is a recently adopted convention for new audio overlays).

File: arch/arm/boot/dts/overlays/Makefile [hunk 3]
```diff
@@ -15,7 +15,8 @@ endif
 dtbo-$(RPI_DT_OVERLAYS) += adau1977-adc.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += ads1015.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += ads7846.dtbo
-dtbo-$(RPI_DT_OVERLAYS) += akkordion-iqdacplus.dtbo
+dtbo-$(RPI_DT_OVERLAYS) += akkordion-iqdacplus.
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/Makefile]: You also need to put back the "dtbo" on this line.

**Comment by clivem** [comment]
@howardqiao This is the third variant of the SB32 driver (or based on the Takazine SB32 driver) I have seen in the past 3 weeks. ESS have been very clear on their position of drivers which operate the DAC in software mode and expose any register programming info! They cannot be Open Source!

I really would suggest that you contact ESS directly to clarify their position over any Open Source driver for the SABRE9018Q2C/K2M.

"I didn't write or publish the driver, the Takazine guys did", isn't a defense. The Takazine guys signed the NDA to obtain the register info. That is a fact! And is what makes anything based on the SB32 drivers, which you have acknowledged in both machine and codec drivers, toxic.

**Comment by howardqiao** [comment]
@clivem Yes,It's based on Takazine SB32 driver.
https://github.com/SatoruKawase/SabreBerry32 
Takazine's code on github of whether open source? Who write or publish the driver on github? ⊙﹏⊙

such as:
https://github.com/miero/botic-dev/blob/1a687859d64838b6efab5920027633a041bf669e/generic/sabre32.c

**Comment by howardqiao** [comment]
@clivem :+1: I pulled this driver under GPL, as this driver (https://github.com/SatoruKawase/SabreBerry32 )written inside the same. Like you said,since the ESS does not endorse this behavior , I still wait for the right time to submit this drive it.what a pity, a great product. so many loyal users.

@pelwell Please delete this pull request,thanks!

**Pull request closed without merging**